### ISSUE TITLE
Can't clone or fetch from private project

### ISSUE THREAD

**Issue by benlindsay** [osfclient/osfclient#144]
I'm sure I'm doing something wrong but I can't figure out what. I have a private project, and I'm able to upload a file with `osf -p abc12 -u me@email.com upload file.txt file.txt`. It also works if I have a `.osfcli.config` file with the project name and username and just type `osf upload file.txt file.txt`. But if I try `osf -p abc12 -u me@email.com fetch file.txt`, or `osf fetch file.txt` in the same directory as an `.osfcli.config` file, or `osf -p abc12 -u me@email.com clone .`, or `osf clone .` in the same directory as an `.osfcli.config` file, I always get the same response of

```
usage: osf fetch [-h] [-f] remote [local]
osf fetch: error: You are not authorized to access this project.
```

When I temporarily switched my project from private to public (I don't have any real info in it yet), I was able to clone it just fine.

I have my password in the `OSF_PASSWORD` environment variable.

Any idea what I'm doing wrong, or what the next steps to diagnose the problem would be?

**Comment by BRosenblatt** [comment]
@ctb and @benlindsay, are there any updates on these issues?

**Comment by benlindsay** [comment]
@BRosenblatt unfortunately none from me yet, if you or anyone else has time to look into them that would be great. I might come back to this within the next month or so. Are you running into issues like this as well?

**Comment by BRosenblatt** [comment]
Thanks, @benlindsay. Not from my side. I run the user support at OSF. The user from this [issue](https://github.com/osfclient/osfclient/issues/159) thinks 159 and 144 are related, and they're having trouble proceeding. Can you all take a look to see what may be happening? And do let us know if the issue is on our end!

**Comment by benlindsay** [comment]
This is on my list of things to get to when I can. Job applications are keeping me busy :) I'll let you know what I find when I get to it.

**Comment by BRosenblatt** [comment]
Great, thank you for the update!

**Comment by keiCetel** [comment]
Hi there, osfclient is a great help but I am afraid I have run into the same issue as @benlindsay. I was wondering whether this has anything to do with the token management (although I am also able to upload data to OSF). I have generated a token but am unclear as to how to make osfclient use it. Can you advise?

(I am using osfclient from an Anaconda prompt on 64-bit Win 7.)

**Comment by raw937** [comment]
I am also having a similar problem. 
I can osf ls. 
But, when I try osf clone
I get ->  osf clone
Please input your password: 
0files [00:07, ?files/s]
usage: osf clone [-h] [output]
osf clone: error: You are not authorized to access this project.

How can a fetch/clone a project?

**Comment by adswa** [comment]
Hey all, I'm having the same problem as well.
When working with a private project, listing the project works fine, cloning or fetching throws ``You are not authorized to access this project`` errors (while fetching fails, it does create an empty file in place of what I would have downloaded), and uploading files works fine.

 Are there by any chance updates on this? Thanks much in advance, and thanks for the osf client!

**Comment by raw937** [comment]
I really love OSF. And, osf-client used to work. But, now it doesn't. 
Its SO SAD. As if we don't make it easy for people it will die. 
I don't want to go back to the SRA.

**Comment by mih** [comment]
I dug through the code and I found this:

Everything works but the download of files. When I compare the URLs used by `osfclient` with the ones advertised by the OSF website I found them to be different:

```
In [45]: f._download_url                                                                     
Out[45]: 'https://osf.io/download/4rpzm/'
```

but OSF website says: `https://osf.io/4rpzm/download`

**Comment by kousu** [comment]
I can vouch for this being the problem. We had to fix it too: https://github.com/neuropoly/spinalcordtoolbox/pull/2671/files

**Comment by mih** [comment]
OK, the fix is trivial. The `File` (`f` in my example below) object internally uses the wrong URL:

```
In [23]: f._download_url                                                                     
Out[23]: 'https://osf.io/4rpzm/download'

In [24]: f._upload_url                                                                       
Out[24]: 'https://files.osf.io/v1/resources/vjnbp/providers/osfstorage/5ee88f2eaa1d010040c4a3b3'
```

When I make them identical, things start working:

```
In [26]: f._download_url = f._upload_url                                                     

In [27]: f.write_to(open('mytest.txt', 'wb'))                                                
100%|█████████████████████████████████████████████████| 4.00/4.00 [00:00<00:00, 11.3kbytes/s]
```

**Comment by felliott** [comment]
Fixed in [v0.0.4])(https://github.com/osfclient/osfclient/tree/0.0.4)
